### PR TITLE
Add support for converting old database flags

### PR DESCRIPTION
**Pull request by youngmit**
Before this, if flags are added to an application, databases from a
previous version would be unreadable, because the Flags serializer makes
sure that the meaning of all flags is the same between the reading app,
and the app that wrote the database. Now, it will only check that all
old flags still exist, but permit converting to the new Flag numeric
values.

### CODE REVIEW

File: armi/bookkeeping/db/database3.py [hunk 1]
```diff
@@ -673,13 +673,23 @@ def loadCS(self):
 
         cs = settings.Settings()
         cs.caseTitle = os.path.splitext(os.path.basename(self.fileName))[0]
-        cs.loadFromString(self.h5db["inputs/settings"][()])
+        try:
+            cs.loadFromString(self.h5db["inputs/settings"][()])
+        except:
```
Comment (ntouran) [inline armi/bookkeeping/db/database3.py:678]: make pylint broad exception exception and explain when it's expected and why it's ok

File: armi/bookkeeping/db/database3.py [hunk 2]
```diff
@@ -673,13 +673,23 @@ def loadCS(self):
...
     def loadBlueprints(self):
         from armi.reactor import blueprints
 
-        bpString = self.h5db["inputs/blueprints"][()]
+        bpString = None
+
+        try:
+            bpString = self.h5db["inputs/blueprints"][()]
+        except:
+            pass
```
Comment (ntouran) [inline armi/bookkeeping/db/database3.py:691]: add explanation and pylint disable

File: armi/reactor/tests/test_composites.py [hunk 3]
```diff
@@ -422,19 +436,36 @@ def test_flagSerialization(self):
...
-            )
+    def test_flagConversion(self):
+        data = [
+            self.TestFlagsA.A,
+            self.TestFlagsA.A | self.TestFlagsA.C,
+            self.TestFlagsA.A | self.TestFlagsA.C | self.TestFlagsA.D,
+        ]
+
+        for f in data:
+            print(int(f))
```
Comment (ntouran) [inline armi/reactor/tests/test_composites.py:454]: still want this in here?

File: armi/reactor/tests/test_composites.py [hunk 4]
```diff
@@ -422,19 +436,36 @@ def test_flagSerialization(self):
...
+
+        serialized, attrs = composites.FlagSerializer._packImpl(data, self.TestFlagsA)
+
+        data2 = composites.FlagSerializer._unpackImpl(
+            serialized, composites.FlagSerializer.version, attrs, self.TestFlagsB
+        )
+        for f in data2:
+            print(int(f))
+
```
Comment (ntouran) [inline armi/reactor/tests/test_composites.py:463]: remove?

File: armi/reactor/tests/test_composites.py [hunk 5]
```diff
@@ -422,19 +436,36 @@ def test_flagSerialization(self):
...
+        expected = [
+            self.TestFlagsB.A,
+            self.TestFlagsB.A | self.TestFlagsB.C,
+            self.TestFlagsB.A | self.TestFlagsB.C | self.TestFlagsB.D,
+        ]
```
Comment (ntouran) [inline armi/reactor/tests/test_composites.py:468]: should be an assert after this checking that result matches

**Review by ntouran — CHANGES_REQUESTED** [review]
* Please add some narrative docs somewhere about how exactly the flag conversion is done.
* Consider updating release notes summarizing the change

File: armi/bookkeeping/db/database3.py (hunk 1 above)
Comment (youngmit) [reply armi/bookkeeping/db/database3.py]: Turns out it should be a KeyError and i was just lazy. Added with comment

File: armi/bookkeeping/db/database3.py (hunk 2 above)
Comment (youngmit) [reply armi/bookkeeping/db/database3.py]: same as above

**Review by ntouran — APPROVED** [review]
sweeet

**Pull request merged**